packages/adp-tooling/src/writer/index.ts

```diff
@@ -0,0 +1,91 @@
+import { join } from 'path';
+import { create as createStorage } from 'mem-fs';
+import type { Editor } from 'mem-fs-editor';
+import { create } from 'mem-fs-editor';
+import type { AdpPreviewConfig, AdpWriterConfig } from '../types';
+import { UI5Config } from '@sap-ux/ui5-config';
+
+/**
+ * Set default values for optional properties.
+ *
+ * @param config configuration provided by the calling middleware
+ * @returns enhanced configuration with default values
+ */
+function setDefaults(config: AdpWriterConfig): AdpWriterConfig {
+    const configWithDefaults: AdpWriterConfig = {
+        app: { ...config.app },
+        target: { ...config.target },
+        ui5: { ...config.ui5 }
+    };
+    configWithDefaults.app.title ??= `Adaptation of ${config.app.reference}`;
+    configWithDefaults.app.layer ??= 'CUSTOMER_BASE';
```

Comment:
For the `manifest.appdescr_variant` if the layer is `CUSTOMER_BASE` then we need to `customer` as prefix to the app id. When the layer is `VENDOR`, then we need to add `vendor` as prefix.

Reply:
Done with https://github.com/SAP/open-ux-tools/pull/1105/commits/778f8c011eac585412b5f6c134a833ddf72f5371

Reply:
Done with https://github.com/SAP/open-ux-tools/pull/1105/commits/67b3f42e07257516be760970c74aef737bf91697